### ISSUE TITLE
Create a proper Remix SDK

### ISSUE THREAD

**Issue by et** [highlight/highlight#5861]

**Comment by et** [comment]
cc @deltaepsilon - will have to pick your brain on the internals of what we did for nextjs. I'll set some time up to chat about this next week.

**Comment by deltaepsilon** [comment]
> cc @deltaepsilon - will have to pick your brain on the internals of what we did for nextjs. I'll set some time up to chat about this next week.

Yes please! I can help with this too. I've set up exactly one Remix POC and I remember none of it.

**Comment by deltaepsilon** [comment]
I'm working on a POC:

https://github.com/highlight/highlight/pull/5992

**Comment by CSFlorin** [comment]
I recommend on the server looking at the recently added `handleError` for error reporting https://github.com/remix-run/remix/releases/tag/remix%401.17.0

**Comment by deltaepsilon** [comment]
> Would be worth doing a bit of research on tracking builds/version in remix (cc @deltaepsilon @et )

I couldn't find any built-in build tracking for Remix referenced in their docs.
